File: LeetCode/1.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:

        outcome = {}
        for i in range(len(nums)) :
            remain = target - nums[i]
            
            if remain in outcome :
                return [outcome[remain], i]
            else :
                outcome[nums[i]] = i

File: LeetCode/1.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer,Integer> indexMap = new HashMap<Integer,Integer>();
        for(int i=0; i<nums.length; i++){
            Integer required = (Integer)(target - nums[i]);
            if(indexMap.containsKey(required)){
                int toReturn[] = {indexMap.get(required), i};
                return toReturn;
            }
            indexMap.put(nums[i], i);
        }
        return null;
    }
}
